test/src/specs/sync/get_blocks.rs

```diff
@@ -0,0 +1,51 @@
+use super::utils::wait_get_blocks;
+use crate::utils::build_headers;
+use crate::{Net, Spec, TestProtocol};
+use ckb_sync::{NetworkProtocol, BLOCK_DOWNLOAD_TIMEOUT};
+use ckb_types::core::HeaderView;
+use log::info;
+use std::thread;
+use std::time::Duration;
+
+pub struct GetBlocksTimeout;
+
+impl Spec for GetBlocksTimeout {
+    crate::name!("get_blocks_timeout");
+
+    crate::setup!(
+        connect_all: false,
+        num_nodes: 2,
+        protocols: vec![TestProtocol::sync()],
+    );
+
+    fn run(&self, mut net: Net) {
+        let node1 = net.nodes.pop().unwrap();
+        let node2 = net.nodes.pop().unwrap();
+        node1.generate_blocks(1);
+        node2.generate_blocks(20);
+
+        let headers: Vec<HeaderView> = (1..=node2.get_tip_block_number())
+            .map(|i| node2.get_header_by_number(i))
+            .collect();
+
+        net.connect(&node1);
+        let (pi, _, _) = net.receive();
+        info!("Send Headers to node1");
+        net.send(NetworkProtocol::SYNC.into(), pi, build_headers(&headers));
+        info!("Receive GetBlocks from node1");
+        assert!(wait_get_blocks(10, &net), "timeout to wait GetBlocks");
+        let wait_get_blocks_secs = 20;
```

Comment:
The BLOCK_DOWNLOAD_TIMEOUT must be greater than 20 seconds, please add this assert.